### ISSUE TITLE
Enable one-qubit optimization for UCCSD

### ISSUE THREAD

**Issue by hhorii** [Qiskit/qiskit-terra#5436]
### What is the expected enhancement?

0.23.1 transpiles a UCCSD circuit without one-qubit optimization though 0.22.0 does with it. Probably deprecating U1, U2, and U3 is a reason.

0.22.0:
```
       ┌─────────────────────┐                                   »
  q_0: ┤ U3(pi/2,pi/2,3pi/2) ├──■────────────────────────────────»
       └──┬───────────────┬──┘┌─┴─┐                              »
  q_1: ───┤ U3(pi/2,0,pi) ├───┤ X ├──■────────────────────────■──»
          └───────────────┘   └───┘┌─┴─┐┌──────────────────┐┌─┴─┐»
  q_2: ────────────────────────────┤ X ├┤ U3(0,0,1.0*θ[0]) ├┤ X ├»
                                   └───┘└──────────────────┘└───┘»
  q_3: ──────────────────────────────────────────────────────────»
                                                                 »
  q_4: ──────────────────────────────────────────────────────────»
                                                                 »
  q_5: ──────────────────────────────────────────────────────────»
                                                                 »
  q_6: ──────────────────────────────────────────────────────────»
                                                                 »
  q_7: ──────────────────────────────────────────────────────────»
                                                                 »
  q_8: ──────────────────────────────────────────────────────────»
                                                                 »
  q_9: ──────────────────────────────────────────────────────────»
                                                                 »
c0: 10/══════════════════════════════════════════════════════════»
                                                                 »
«                        ┌────────────────────┐                               »
«  q_0: ────────■────────┤ U3(pi/2,3pi/2,-pi) ├──■─────────────────────────■──»
«             ┌─┴─┐      └┬──────────────────┬┘┌─┴─┐┌───────────────────┐┌─┴─┐»
«  q_1: ──────┤ X ├───────┤ U3(pi/2,0,3pi/2) ├─┤ X ├┤ U3(0,0,-1.0*θ[0]) ├┤ X ├»
«       ┌─────┴───┴─────┐ └──────────────────┘ └───┘└───────────────────┘└───┘»
«  q_2: ┤ U3(pi/2,0,pi) ├─────────────────────────────────────────────────────»
«       └───────────────┘                                                     »
«  q_3: ──────────────────────────────────────────────────────────────────────»
«                                                                             »
«  q_4: ──────────────────────────────────────────────────────────────────────»
«                                                                             »
«  q_5: ──────────────────────────────────────────────────────────────────────»
«                                                                             »
«  q_6: ──────────────────────────────────────────────────────────────────────»
«                                                                             »
«  q_7: ──────────────────────────────────────────────────────────────────────»
«                                                                             »
«  q_8: ──────────────────────────────────────────────────────────────────────»
«                                                                             »
«  q_9: ──────────────────────────────────────────────────────────────────────»
«                                                                             »
«c0: 10/══════════════════════════════════════════════════════════════════════»
```
0.23.0
```
       ┌───────────┐┌─────────────────┐                                  »
  q_0: ┤ U3(π,0,π) ├┤ U(π/2,-π/2,π/2) ├──■───────────────────────────────»
       └┬─────────┬┘└─────────────────┘┌─┴─┐                             »
  q_1: ─┤ U2(0,π) ├────────────────────┤ X ├──■───────────────────────■──»
        └─────────┘                    └───┘┌─┴─┐┌─────────────────┐┌─┴─┐»
  q_2: ─────────────────────────────────────┤ X ├┤ U(0,0,1.0*θ[0]) ├┤ X ├»
                                            └───┘└─────────────────┘└───┘»
  q_3: ──────────────────────────────────────────────────────────────────»
                                                                         »
  q_4: ──────────────────────────────────────────────────────────────────»
                                                                         »
  q_5: ──────────────────────────────────────────────────────────────────»
                                                                         »
  q_6: ──────────────────────────────────────────────────────────────────»
                                                                         »
  q_7: ──────────────────────────────────────────────────────────────────»
                                                                         »
  q_8: ──────────────────────────────────────────────────────────────────»
                                                                         »
  q_9: ──────────────────────────────────────────────────────────────────»
                                                                         »
c0: 10/══════════════════════════════════════════════════════════════════»
                                                                         »
«                  ┌──────────────────┐    ┌─────────┐         »
«  q_0: ─────■─────┤ U(-π/2,-π/2,π/2) ├────┤ U2(0,π) ├──────■──»
«          ┌─┴─┐   └───┬─────────┬────┘┌───┴─────────┴───┐┌─┴─┐»
«  q_1: ───┤ X ├───────┤ U2(0,π) ├─────┤ U(π/2,-π/2,π/2) ├┤ X ├»
«       ┌──┴───┴──┐    └─────────┘     └─────────────────┘└───┘»
«  q_2: ┤ U2(0,π) ├────────────────────────────────────────────»
«       └─────────┘                                            »
«  q_3: ───────────────────────────────────────────────────────»
«                                                              »
«  q_4: ───────────────────────────────────────────────────────»
«                                                              »
«  q_5: ───────────────────────────────────────────────────────»
«                                                              »
«  q_6: ───────────────────────────────────────────────────────»
«                                                              »
«  q_7: ───────────────────────────────────────────────────────»
«                                                              »
«  q_8: ───────────────────────────────────────────────────────»
«                                                              »
«  q_9: ───────────────────────────────────────────────────────»
«                                                              »
«c0: 10/═══════════════════════════════════════════════════════»
```

Related PRs: #5429, #5431

**Comment by Cryoris** [comment]
Since #5429 is merged, is this problem resolved? 🙂

**Comment by mtreinish** [comment]
Not quite yet. That was only part of the issue UCCSD with aer exposes. It would have been sufficient without https://github.com/Qiskit/qiskit-aqua/commit/4cca284cd4cf134137d4a5c8246008d197e39888 and everything was still in U. But, with that aqua change that and aer's overcomplete basis set we need to merge #5436 too, and also adjust the selection logic in preset pass managers. For example: https://github.com/Qiskit/qiskit-terra/blob/master/qiskit/transpiler/preset_passmanagers/level1.py#L178-L182 doesn't work for aer because it we don't simplify chains of other single qubit gates because the basis translator doesn't translate most gates for aer.

**Comment by Avhijit-codeboy** [comment]
This issue has been resolved with the merging of #5429 and #5431.
